From 827ecabb918e3d6e545bec2edb27afe50ced8a70 Mon Sep 17 00:00:00 2001
From: jinghua <xigu@marvell.com>
Date: Sat, 13 Aug 2016 00:37:16 +0800
Subject: [PATCH 0927/2241] arm64: dts: a3700: enable hw iocc for all io units

- Armada3700 A0 chip supports HW IOCC, so add 'dma-coherent' for all
  IO units that supporting coherent DMA operations to enable it on them.
- Requires setting address decoding with shareable attributes in u-boot for
  relevant units: PCIe, GbE0, GbE1, SD, eMMC, SATA, USB2, USB3 and XOR.
- HW IOCC is not supported on Z1 chip, so dma-coherent could not be added
  to dt nodes in armada37xx.dtsi, since armada37xx_rz.dtsi inheritates
  everything from it.
- This patch adds dma-coherent to armada372x.dtsi and armada371x.dtsi,
  which every 37xx DB dts file inheritates. When Z1 chip support is
  removed later, these dma-coherent property should be moved to
  armada37xx.dtsi.

Change-Id: Ic7b071ee037eb07533cb5fdf872714d149ea6878
Signed-off-by: jinghua <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/31905
Tested-by: Wilson Ding <dingwei@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
(cherry picked from commit ff84c1d28775dbc196cd2553fd99ffe5ef1a0018)
Reviewed-on: http://vgitil04.il.marvell.com:8080/32414
Tested-by: Star_Automation <star@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-371x.dtsi | 34 ++++++++++++++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-372x.dtsi | 33 +++++++++++++++++++++++++++
 2 files changed, 67 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-371x.dtsi b/arch/arm64/boot/dts/marvell/armada-371x.dtsi
index c9e5325..486ed82 100644
--- a/arch/arm64/boot/dts/marvell/armada-371x.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-371x.dtsi
@@ -50,4 +50,38 @@
 / {
 	model = "Marvell Armada 3710 SoC";
 	compatible = "marvell,armada3710", "marvell,armada3700";
+	soc {
+		internal-regs {
+			eth0: ethernet@30000 {
+				   dma-coherent;
+			};
+			eth1: ethernet@40000 {
+				dma-coherent;
+			};
+			usb3@58000 {
+				dma-coherent;
+			};
+			usb@5e000 {
+				dma-coherent;
+			};
+			xor@60900 {
+				dma-coherent;
+			};
+			sdhci1: sdhci@d0000 {
+				dma-coherent;
+			};
+
+			sdhci0: sdhci@d8000 {
+				dma-coherent;
+			};
+			sata@e0000 {
+				dma-coherent;
+			};
+			pcie-controller {
+				pcie@1,0 {
+					dma-coherent;
+				};
+			};
+		};
+	};
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-372x.dtsi b/arch/arm64/boot/dts/marvell/armada-372x.dtsi
index f292a00..fa2fd89 100644
--- a/arch/arm64/boot/dts/marvell/armada-372x.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-372x.dtsi
@@ -59,5 +59,38 @@
 			enable-method = "psci";
 		};
 	};
+	soc {
+		internal-regs {
+			eth0: ethernet@30000 {
+				   dma-coherent;
+			};
+			eth1: ethernet@40000 {
+				dma-coherent;
+			};
+			usb3@58000 {
+				dma-coherent;
+			};
+			usb@5e000 {
+				dma-coherent;
+			};
+			xor@60900 {
+				dma-coherent;
+			};
+			sdhci1: sdhci@d0000 {
+				dma-coherent;
+			};
 
+			sdhci0: sdhci@d8000 {
+				dma-coherent;
+			};
+			sata@e0000 {
+				dma-coherent;
+			};
+			pcie-controller {
+				pcie@1,0 {
+					dma-coherent;
+				};
+			};
+		};
+	};
 };
-- 
2.7.4

